File: W1D5-AndroidView/app/src/main/java/com/obasoglu/android/androidviews/MainActivity.java
package com.obasoglu.android.androidviews;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button okButton;
    private ImageView mainImage;

    private TextView rakamText;

    private int toplamRakam = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
    }

    private void initViews() {
        // Access views
        okButton = findViewById(R.id.ok_button);
        mainImage = findViewById(R.id.image_main);
        rakamText = findViewById(R.id.text_content);
        //  Change image resource. First layout resource set in attribute, later this
        // function changes it.
        mainImage.setImageResource(R.drawable.background_wild);

        okButton.setOnClickListener(new View.OnClickListener() {
            // Will be triggered when user clicks on "okButton"
            @Override
            public void onClick(View view) {
                Log.d("MainActivity", "Button is clicked.");
                toplamRakam++;
                String toplamRakamText = String.valueOf(toplamRakam);
                rakamText.setText(toplamRakamText);
                Toast.makeText(MainActivity.this, toplamRakamText , Toast.LENGTH_SHORT).show();
            }
        });


    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }
}

File: W16D1-Generics/app/src/main/java/com/tinyfalcon/w16d1_generics/MainActivity.kt
package com.tinyfalcon.w16d1_generics

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val a = returnsInt(10)
        val b = returnsString("hello")
        val square: Int = a * a

        val aa = handleInput(10)
        val bb = handleInput("hello")
        // val square2: Int = aa * aa  compile error

        val aaa = handleInputGeneric(10)
        val bbb = handleInputGeneric("hello")
        val square3: Int = aaa * aaa
    }

    private fun <T> handleInputGeneric(param: T): T {
        return param
    }

    private fun handleInput(param: Any): Any? {
        val returnValue = when(param) {
            is Int -> returnsInt(param)
            is String -> returnsString(param)
            else -> null
        }
        return returnValue
    }

    private fun returnsInt(param: Int): Int {
        return param
    }

    private fun returnsString(param: String): String {
        return param
    }
}